src/x/serialize/decoder_test.go

```diff
@@ -179,6 +182,26 @@ func TestDecodeLiteralTooLong(t *testing.T) {
 	require.Error(t, d.Err())
 }
 
+func TestDecodeLiteralOfMaximumPossibleLength(t *testing.T) {
+	name := strings.Repeat("n", math.MaxUint16)
+	value := strings.Repeat("v", math.MaxUint16)
```

Comment:
Why not use `DefaultMaxTagLiteralLength` instead of `math.MaxUint16`? Because if someone increases `DefaultMaxTagLiteralLength`, then this test no longer does what it says.